### ISSUE TITLE
SailsJs. From controller "update" to "beforeValidate" values change. Why?

### ISSUE THREAD

**Issue by ginolon** [balderdashy/sails#2294]
When the object "userObj" pass from "beforeValidate" in UserModel.js the values change.

I want to understand why?

The code in UserController.js

    update: function(req, res, next) {
    
          if (req.session.User.admin) {
          var userObj = {
            name: req.param('name'),
            title: req.param('title'),
            email: req.param('email'),
            admin: req.param('admin')
          }
            } else {
          var userObj = {
            name: req.param('name'),
            title: req.param('title'),
            email: req.param('email')
          }
          }
    
          /*var userObj = req.params.all();
          if (!req.session.User.admin) delete userObj.admin;
          */
    
          console.log("req.param(admin) before User.update: "+req.param('admin')+"\n");
    
          console.log("userObj.admin before User.update: "+userObj.admin+"\n");
    
        User.update(req.param('id'), userObj).exec(function userUpdated (err) {
    
          if (err) {
            return res.redirect('/user/edit/' + req.param('id'));
          }
    
          res.redirect('/user/show/' + req.param('id'));
        });
      	},

**The two `console.log` before `User.update` are the same: `yes`. But when they pass from `beforeValidate` they change to `false`.**

The code from User.js (in model):

    beforeValidate: function (values, next) {
    
    //console.log(values.admin);
    //console.log(values.admin[1]);    

    /*if (typeof values.admin !== 'undefined') {
      if (values.admin === 'unchecked') {
        values.admin = false;
      } else if (values.admin[1] === 'on') {
        values.admin = true;
      }
    }*/
    console.log("values.admin: "+values.admin);

    if (values.admin == "yes") {
        values.admin = true;
      } else {
        values.admin = false;
    }

    console.log("values.admin: "+values.admin+"\n");

    next();
    },

**The two `console.log` before and after `if else` are the same: `false`. But when they come from `update` in the Controller they are `yes`.**

The code from edit.ejs:

    <form action="/user/update/<%= user.id %>" method="POST" class="form-signin">
		<h2>Modify</h2>
		<input value="<%= user.name %>" name="name" type="text" class="form-control"/>
		<input value="<%= user.title %>" name="title" type="text" class="form-control"/>
		<input value="<%= user.email %>" name="email" type="text" class="form-control"/>
		<% if (session.authenticated && session.User.admin) { %>
			<% if (user.admin) { %>
				<label class="checkbox"><input type="checkbox" name="admin" checked value="yes"> Admin</label>
			<% } else { %>
				<label class="checkbox"><input type="checkbox" name="admin" value="yes"> Admin</label>
			<% } %> 
		<% } %>
		<input type="submit" value="Save" class="btn btn-lg btn-primary btn-block"/>
		<input type="hidden" name="_csrf" value="<%= _csrf %>" />
	</form>

**Comment by CWyrtzen** [comment]
@ginolon @motephyr did either of you ever find an answer to your question?

**Comment by sailsbot** [comment]
Thanks for posting, @ginolon.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
